src/lib.rs

```diff
@@ -422,17 +430,50 @@ impl Mock {
 
         true
     }
+
+    fn body_matches(&self, request: &mut Request) -> bool {
+        if !self.request.body.is_empty() {
+            let mut body = String::new();
+            request.read_to_string(&mut body).unwrap();
+
+            let actual_json: Value = serde_json::from_str(&body).unwrap();
```

Comment:
I'm planning on making this more generic to support multiple response content types.